app/uk/gov/hmrc/helptosavefrontend/models/HTSEvent.scala

```diff
@@ -76,13 +75,27 @@ case class ApplicationSubmittedEvent(auditSource: String, userInfo: NSIUserInfo)
   )
 }
 
-case class EligibilityCheckEvent(auditSource: String, nino: String, errorDescription: Option[String])(implicit hc: HeaderCarrier) extends HTSEvent {
+case class EligibilityResult(nino: NINO, errorDescription: Option[String])(implicit hc: HeaderCarrier) extends HTSEvent {
   val value: DataEvent = HTSEvent(
-    auditSource, "eligibilityCheck", {
-    val basicMap = Map[String, String]("nino" -> nino)
-    errorDescription.fold(basicMap + ("eligible" -> "true")) { reason ⇒
-      basicMap + ("eligible" -> "false") + ("reason" -> reason)
+    "EligibilityResult", {
+      val basicMap = Map[String, String]("nino" -> nino, "eligible" -> "true")
+      errorDescription.fold(basicMap) { reason ⇒
+        basicMap + ("eligible" -> "false") + ("reason" -> reason)
+      }
     }
-  }
+  )
+}
+
+case class EmailChanged(nino: NINO, oldEmail: String, newEmail: String)(implicit hc: HeaderCarrier) extends HTSEvent {
+  val value: DataEvent = HTSEvent(
+    "EmailChanged",
+    Map[String, String]("nino" -> nino, "oldEmail" -> oldEmail, "newEmail" -> newEmail)
+  )
+}
+
+case class SuspiciousActivity(activity: String)(implicit hc: HeaderCarrier) extends HTSEvent {
```

Comment:
This doesn't quite match up with the format agreed at the bottom of this page: 

https://confluence.tools.tax.service.gov.uk/display/TM/Help+to+Save+Service+-+Transaction+Monitoring+Assessment

You are also missing a couple of instances where we need to send a `SuspiciousActivity` event (missing_enrolment|missing_email_record|malformed_redirect)

Reply:
fixed

Reply:
we're missing events for the case in the email verification journey where:
- the continue URL parameter can't be decrypted (malformed_redirect)
- a user is updating their email but they aren't in our enrolment mongo store
- a user is updating their email but they're not in our email mongo store

I can show you where those points are if you like